src/coap_client_zephyr.c

```
@@ -0,0 +1,1072 @@
+            LOG_DBG("Handle GET %s", req->path);
+            err = golioth_coap_req_cb(req->client, COAP_METHOD_GET,
+                                      PATHV(req->path_prefix, req->path),
+                                      golioth_content_type_to_coap_format(req->get.content_type),
+                                      NULL, 0,
+                                      golioth_coap_cb, req,
+                                      0);
+            break;
+        case GOLIOTH_COAP_REQUEST_GET_BLOCK:
+            LOG_DBG("Handle GET_BLOCK %s", req->path);
+            err = golioth_coap_get_block(req);
+            break;
+        case GOLIOTH_COAP_REQUEST_POST:
+            LOG_DBG("Handle POST %s", req->path);
+            err = golioth_coap_req_cb(req->client, COAP_METHOD_POST,
+                                      PATHV(req->path_prefix, req->path),
+                                      golioth_content_type_to_coap_format(req->post.content_type),
+                                      req->post.payload, req->post.payload_size,
+                                      golioth_coap_cb, req,
+                                      0);
+            golioth_sys_free(req->post.payload);
+            break;
+        case GOLIOTH_COAP_REQUEST_DELETE:
+            LOG_DBG("Handle DELETE %s", req->path);
+            err = golioth_coap_req_cb(req->client, COAP_METHOD_DELETE,
+                                      PATHV(req->path_prefix, req->path),
+                                      COAP_CONTENT_FORMAT_APP_OCTET_STREAM /* not used */,
+                                      NULL, 0,
+                                      golioth_coap_cb, req,
+                                      0);
+            break;
+        case GOLIOTH_COAP_REQUEST_OBSERVE:
+            LOG_DBG("Handle OBSERVE %s", req->path);
+            err = golioth_coap_observe(req, client);
+            break;
+        default:
+            LOG_WRN("Unknown request_msg type: %u", req->type);
```

Can we set `status` here so we can return an error to the caller?

---

Yes, definitely. See https://github.com/golioth/golioth-firmware-sdk/compare/00f2b0e96d95b498ba28e0ced956a195b03d6997..48cbd0a5dec09c5557a0c623ace897ded7bc5502